### PR TITLE
remove all dep code for 3.1

### PR DESCRIPTION
**Pull request by nabobalis**
I always forgot the min cycle for items.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -63,6 +59,7 @@ net =
   drms>=0.6.1
   pandas>=0.24.0
   python-dateutil>=2.8.0
+  sunkit_instruments>=0.2.0
```
Comment (nabobalis) [inline setup.cfg:62]: We use one function from this for net. Maybe we should copy it here.

**Review by dstansby — CHANGES_REQUESTED** [review]
This still needs individual changelogs for each removal specifying what the alternative is to what has been removed.

**Comment by dstansby** [comment]
sorry, just noticed this is draft...

**Comment by nabobalis** [comment]
> This still needs individual changelogs for each removal specifying what the alternative is to what has been removed.

Changelogs added for you to review.

**Comment by nabobalis** [comment]
Now to consider if we want to silence, `W504 line break after binary operator`. Maybe we want the line to be broke before a binary operator.

File: changelog/5304.removal.1.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+``sunpy.roi.chaincode.Chaincode`` has been removed in favour of `sunpy.net.helio.Chaincode`.
```
Comment (Cadair) [inline changelog/5304.removal.1.rst:1]: In general I suggest you add "the deprecated x has been removed" just so it's clear to people not paying attention we gave everyone warning.

Comment (nabobalis) [reply changelog/5304.removal.1.rst:1]: K

**Review by wtbarnes — APPROVED** [review]

**Comment by wtbarnes** [comment]
Test failures in `net` are real,

```python-traceback
2021-05-04T17:28:39.4095315Z =================================== FAILURES ===================================
2021-05-04T17:28:39.4096409Z _________________________ test_tables_single_response __________________________
2021-05-04T17:28:39.4098123Z [gw2] linux -- Python 3.8.9 /home/vsts/work/1/s/.tox/py38-online/bin/python
2021-05-04T17:28:39.4098982Z 
2021-05-04T17:28:39.4099727Z     @pytest.mark.remote_data
2021-05-04T17:28:39.4100535Z     def test_tables_single_response():
2021-05-04T17:28:39.4101355Z         results = Fido.search(
2021-05-04T17:28:39.4102380Z             a.Time("2012/1/1", "2012/1/5"), a.Instrument.lyra, a.Level.two)
2021-05-04T17:28:39.4103225Z     
2021-05-04T17:28:39.4104121Z         with pytest.warns(SunpyDeprecationWarning):
2021-05-04T17:28:39.4104948Z >           tables = results.tables
2021-05-04T17:28:39.4106118Z E           AttributeError: 'UnifiedResponse' object has no attribute 'tables'
2021-05-04T17:28:39.4106850Z 
2021-05-04T17:28:39.4108406Z /home/vsts/work/1/s/.tox/py38-online/lib/python3.8/site-packages/sunpy/net/tests/test_fido.py:205: AttributeError
2021-05-04T17:28:39.4109469Z _______________________________ test_time_query ________________________________
2021-05-04T17:28:39.4110678Z [gw3] linux -- Python 3.8.9 /home/vsts/work/1/s/.tox/py38-online/bin/python
2021-05-04T17:28:39.4111495Z 
2021-05-04T17:28:39.4113330Z client = <sunpy.net.helio.hec.HECClient object at 0x7ff832c1c3a0>
2021-05-04T17:28:39.4114276Z sunpy.net.helio.hec.HECClient
2021-05-04T17:28:39.4114898Z 
2021-05-04T17:28:39.4115777Z Provides access to the HELIO webservices.
2021-05-04T17:28:39.4116737Z 
2021-05-04T17:28:39.4117574Z Attr Type Name Description
2021-05-04T17:28:39.4118581Z --------- ---- -----------
2021-05-04T17:28:39.4119246Z 
2021-05-04T17:28:39.4119951Z     @pytest.mark.remote_data
2021-05-04T17:28:39.4120876Z     def test_time_query(client):
2021-05-04T17:28:39.4122038Z         start = '2005/01/03'
2021-05-04T17:28:39.4124004Z         end = '2005/12/03'
2021-05-04T17:28:39.4126354Z         table_name = 'rhessi_hxr_flare'
2021-05-04T17:28:39.4126987Z         with pytest.raises(SunpyDeprecationWarning):
2021-05-04T17:28:39.4127535Z >           res = client.time_query(start, end, table=table_name, max_records=10)
2021-05-04T17:28:39.4128329Z E           AttributeError: 'HECClient' object has no attribute 'time_query'
2021-05-04T17:28:39.4128715Z 
2021-05-04T17:28:39.4129451Z ../../.tox/py38-online/lib/python3.8/site-packages/sunpy/net/helio/tests/test_helio.py:285: AttributeError
2021-05-04T17:28:39.4130116Z ________________________ test_tables_multiple_response _________________________
2021-05-04T17:28:39.4130896Z [gw2] linux -- Python 3.8.9 /home/vsts/work/1/s/.tox/py38-online/bin/python
2021-05-04T17:28:39.4131276Z 
2021-05-04T17:28:39.4131653Z     @pytest.mark.remote_data
2021-05-04T17:28:39.4132081Z     def test_tables_multiple_response():
2021-05-04T17:28:39.4132831Z         results = Fido.search(a.Time('2012/3/4', '2012/3/6'),
2021-05-04T17:28:39.4133463Z                               a.Instrument.lyra | (a.Instrument.rhessi & a.Physobs.summary_lightcurve))
2021-05-04T17:28:39.4133944Z     
2021-05-04T17:28:39.4134378Z         with pytest.warns(SunpyDeprecationWarning):
2021-05-04T17:28:39.4134837Z >           tables = results.tables
2021-05-04T17:28:39.4135551Z E           AttributeError: 'UnifiedResponse' object has no attribute 'tables'
2021-05-04T17:28:39.4135950Z 
2021-05-04T17:28:39.4136906Z /home/vsts/work/1/s/.tox/py38-online/lib/python3.8/site-packages/sunpy/net/tests/test_fido.py:220: AttributeError
2021-05-04T17:28:39.4137615Z _____________________________ test_search_metadata _____________________________
2021-05-04T17:28:39.4138402Z [gw3] linux -- Python 3.8.9 /home/vsts/work/1/s/.tox/py38-online/bin/python
2021-05-04T17:28:39.4138785Z 
2021-05-04T17:28:39.4139212Z client = <sunpy.net.jsoc.jsoc.JSOCClient object at 0x7ff833b1b790>
2021-05-04T17:28:39.4139890Z sunpy.net.jsoc.jsoc.JSOCClient
2021-05-04T17:28:39.4140171Z 
2021-05-04T17:28:39.4140654Z Provides access to the JSOC ...                                                                                                     
2021-05-04T17:28:39.4141510Z Length = 507 rows
2021-05-04T17:28:39.4141766Z 
2021-05-04T17:28:39.4142147Z     @pytest.mark.remote_data
2021-05-04T17:28:39.4142596Z     def test_search_metadata(client):
2021-05-04T17:28:39.4143117Z         with pytest.raises(SunpyDeprecationWarning):
2021-05-04T17:28:39.4144039Z >           metadata = client.search_metadata(a.Time('2020-01-01T00:00:00', '2020-01-01T00:02:00'),
2021-05-04T17:28:39.4145165Z                                               a.jsoc.Series('aia.lev1_euv_12s'), a.Wavelength(304*u.AA))
2021-05-04T17:28:39.4146043Z E           AttributeError: 'JSOCClient' object has no attribute 'search_metadata'
2021-05-04T17:28:39.4146434Z 
2021-05-04T17:28:39.4147191Z ../../.tox/py38-online/lib/python3.8/site-packages/sunpy/net/jsoc/tests/test_jsoc.py:302: AttributeError
```

**Review by wtbarnes — CHANGES_REQUESTED** [review]
The `net` tests need to be adjusted for the methods/attributes that are being removed here (see failures). Also, `docs/code_ref/roi.rst` needs to be removed as it is causing the docs build to fail.

**Comment by nabobalis** [comment]
Hopefully should be fixed now.

**Comment by Cadair** [comment]
Still busted.

**Comment by nabobalis** [comment]
```
FAILED ../../.tox/py38-online/lib/python3.8/site-packages/sunpy/net/jsoc/tests/test_jsoc.py::test_build_table
FAILED ../../.tox/py38-online/lib/python3.8/site-packages/sunpy/net/tests/test_fido.py::test_tables_single_response
FAILED ../../.tox/py38-online/lib/python3.8/site-packages/sunpy/net/tests/test_fido.py::test_tables_multiple_response
FAILED ../../.tox/py38-online/lib/python3.8/site-packages/sunpy/net/jsoc/tests/test_jsoc.py::test_lookup_records_errors
```
NO IT AINT

**Comment by Cadair** [comment]
boooo JSOC booo

**Review by Cadair — APPROVED** [review]

**Review by dstansby — CHANGES_REQUESTED** [review]
The online test failures look like they need fixing (as opposed to being servers down)

**Comment by nabobalis** [comment]
Oh yeah so they are.

**Comment by nabobalis** [comment]
Fingers crossed, this time.

File: sunpy/net/jsoc/jsoc.py [hunk 3]
```diff
@@ -748,64 +739,52 @@ def _lookup_records(self, iargs):
         """
         Do a LookData request to JSOC to workout what results the query returns.
         """
-
         isMeta = iargs.get('meta', False)
```
Comment (nabobalis) [inline sunpy/net/jsoc/jsoc.py:742]: I wonder if this needs removing?

File: sunpy/net/jsoc/jsoc.py [hunk 4]
```diff
@@ -823,12 +802,6 @@ def _lookup_records(self, iargs):
 
         r = c.query(ds, key=key, rec_index=isMeta)
 
-        # If the method was called from search_metadata(), return a Pandas Dataframe,
-        # otherwise return astropy.table
-        # TODO: this check should also be removed post 3.0
-        if isMeta:
-            return r
-
         if r is None or r.empty:
```
Comment (nabobalis) [inline sunpy/net/jsoc/jsoc.py:805]: I do wonder if this entire block needs fixing.

File: changelog/5304.removal.6.rst [hunk 5]
```diff
@@ -0,0 +1,8 @@
+Removed various deprecated methods on our Fido clients and responses:
+
+1. ``UnifiedResponse.build_table``, ``UnifiedResponse.tables``, ``UnifiedResponse.responses``, ``UnifiedResponse.get_response`` and ``UnifiedResponse.blocks`` as `UnifiedResponse` is now an `astropy.table.Table` that is sliceable.
+2. ``UnifiedResponse.response_block_properties`` as `UnifiedResponse.path_format_keys` was added as a better replacement.
+3. ``HECClient.time_query`` as you can now use ``Fido.search`` directly.
+4. ``sunpy.net.jsoc.attrs.Keys`` was unnecessary.
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply changelog/5304.removal.6.rst:6]: Yeah this is more accurate.

**Review by wtbarnes — APPROVED** [review]

File: changelog/5304.removal.rst [hunk 6]
```diff
@@ -0,0 +1 @@
+The deprecated ``sunpy.map.GenericMap.draw_rectangle()`` has been removed, the replacement is `sunpy.map.GenericMap.draw_quadrangle`
```
Comment (ayshih) [inline changelog/5304.removal.rst:1]:
```suggestion
The deprecated ``sunpy.map.GenericMap.draw_rectangle()`` has been removed, the replacement is :meth:`sunpy.map.GenericMap.draw_quadrangle`
```
=)

**Pull request merged**